### DISCUSSION TITLE
Не запускается новый zapret

### DISCUSSION THREAD

**Discussion by Uefi1** [bol-van/zapret#874] [Q&A]
Привет не запускается новый zapret winws.exe окно сразу произвольно закрывается при старте а старый работает

**Comment by bol-van** [comment]
А поподробнее ? Какие опции командной строки ? Через что запускается ?

**Comment by Uefi1** [comment]
> А поподробнее ? Какие опции командной строки ? Через что запускается ?

Опции не причём тут просто в старом есть такая директория:
zapret-master\binaries\win64\zapret-winws\winws.exe
А в новом только такая:
zapret-master\binaries\win64\winws.exe

И winws.exe в старой версии из этой директории zapret-master\binaries\win64\ тоже не работает только из этой zapret-master\binaries\win64\zapret-winws\winws.exe
Я уж не знаю чем они отличаются и почему их два

**Comment by bol-van** [comment]
Тот вариант остался только в bundle
Из этого мало взять только лишь winws.exe 
Там появился файл cygwin1.dll, которого раньше не было

**Comment by bol-van** [comment]
Почему их 2 было написано в docs/windows.txt

> 
> Существует неочевидный момент, каcаемый запуска winws из cygwin шелла. Если в директории, где находится winws, находится копия cygwin1.dll, winws не запустится.
> Если нужен запуск под cygwin, то следует удалить или переместить cygwin1.dll из binaries/win64. Это нужно для работы blockcheck.
> 

А если не из cygwin - cygwin1.dll нужен там, где winws

**Comment by Uefi1** [comment]
> А если не из cygwin - cygwin1.dll нужен там, где winws

Ну и где я его высру если в новой нету ?

**Comment by bol-van** [comment]
Если будем общаться в такой манере, я щас закрою issue и байбай
В последней версии файлы cygwin1.dll помещены в win32 и win64
См дерево проекта

**Comment by RiseOfDeath** [comment]
К данной проблеме могу лишь добавить что у меня все работало, но в один прекрасный день на одном из компов неожиданно касперский начал ругаться на WinDivert (как на легальное ПО, которое может использоваться бла-бла-бла), причем на другом все работает и касперскому пофиг. (И да, на обоих десятка).

Даже если добавить WinDivert в исключения, то все равно приложение сразу закрывается. Если запустить blockcheck то вываливаются ошибки типа такого
```
...
- checking winws --wf-l3=ipv4 --wf-tcp=80 --hostcase
windivert: error opening filter: The system cannot find the path specified.
suspicious redirection 307 to : http://freedom-vrn.ru/support/block.html
UNAVAILABLE
- checking winws --wf-l3=ipv4 --wf-tcp=80 --hostspell=hoSt
windivert: error opening filter: The system cannot find the path specified.
suspicious redirection 307 to : http://freedom-vrn.ru/support/block.html
UNAVAILABLE
...
```
Обратите внимание на сообщение windivert: error opening filter: The system cannot find the path specified. (причем оно в логи не попадает).

Также замечу что тоже самое поведение наблюдается на "проблемном" компе с тем же GoodbyDPI и еще одним аналогичным средством на основе WinDivert
Увы, сказать что ничего не менялось не могу, т.к. проблемным компом пользуется жена, так что если что и было изменено - я врядли смогу получить какое-то внятное пояснение.

**Comment by bol-van** [comment]
С антивирусами давно уже выяснили. С ними надо разбираться не разрабам, а пользователям.
Потому что проблема не в софте, проблема именно в антивирусе.
И это к данному топику вообще не имеет отношения.

**Comment by RiseOfDeath** [comment]
> И это к данному топику вообще не имеет отношения.

Дело в том, что на "проблемном" компе проблема не исчезает даже если вообще убрать антивирус (уже проверил)
Собственно сама программа просто закрывается молча (т.е. так же, как у @Uefi1, если я правильно понял что он написал), а blockcheck все так же спамит ошибкой windivert: error opening filter: The system cannot find the path specified.

**Comment by bol-van** [comment]
Под администратором надо выполнить
```
sc delete windivert
sc stop windivert
```
И не запускать прогу из директории неродной битности.
Если система 64, то из 32 не будет работать.
Так же не запускать с различных сетевых дисков и дисков, которые имеют букву только в пространстве пользователя в его терминальной сессии.

**Comment by RiseOfDeath** [comment]
sc stop windivert говорит что сервис не запущен.

Что касается запуска, то я запускаю zapret-winws/preset_russia.cmd (архив https://github.com/bol-van/zapret-win-bundle/archive/refs/heads/master.zip просто распакован с корень системного диска)

**Comment by bol-van** [comment]
sc delete еще надо

sc qc windivert 
покажет в каком месте сейчас он ищет windivert64.sys

**Comment by RiseOfDeath** [comment]
> sc qc windivert

Во, точно. Спасибо, помогло. Он не там искал, после sc delete windivert все заработало.

**Comment by Uefi1** [comment]
> sc delete еще надо
> 
> sc qc windivert покажет в каком месте сейчас он ищет windivert64.sys

Прошу прощения за мою небольшую грубость ранее, я знаешь что заметил за Win7 после перезагрузки zapret не запускается приходится делать:
sc stop windivert
sc delete windivert

А после я windivert.dll и windivert64.dll заменяю в запрете от GoodByeDpi и с ними уже нормально корректно работает, сделай пожалуйста чтобы windvert из коробки поставлялся от GoodByeDPI чтобы удобно было и так же cygwin1.dll положи в новую версию кому не нужно его удалят и ещё не забывай добавлять
tls_clienthello_www_google_com.bin и quic_initial_www_google_com.bin из коробки

**Comment by Uefi1** [comment]
И ещё вот одна проблема могу ли я добавить пропись в winws.exe чтобы параметры загружались из config, например я хочу поставить ярлык на рабочий стол и в нём прописать "\winws.exe" -config myparams.cfg ну как то вот так
а сейчас у меня параметры в cmd или bat что вообщем то не особо то нравится антивирусам такие файлы

**Comment by bol-van** [comment]
Пока не реализовано.
Если антивирус жалуется на простые батники, то он сам по себе является большой проблемой, от которой стоило бы избавиться.

**Comment by bol-van** [comment]
>знаешь что заметил за Win7 после перезагрузки zapret не запускается приходится делать:
> sc stop windivert
> sc delete windivert

sc qc windivert покажет где он сейчас находится
если не в фолдере с winws, это повод разобраться почему. может GDPI тоже в автозапуске или в службе ?
Вместе они плохо будут уживаться.
В последней версии bundle добавлен windivert_delete.cmd для упрощения разборок с виндивертом.
Он сначала покажет где он, потом спросит надо ли удалить.

cygwin1.dll лежит в основной репе. Если после скачивания его там нет, это может опять быть пинок в сторону бестолкового антивируса.

Что касается версии, win7 сейчас уже мало используется, потому не вижу смысла предлагать по умолчанию устаревшую версию компоненты. Пусть лучше копируют, это описано в docs/windows.txt. Или накатят пак от симпликса с мерами предосторожности

**Comment by Uefi1** [comment]
> Пока не реализовано.
> Если антивирус жалуется на простые батники, то он сам по себе является большой проблемой, от которой стоило бы избавиться.

Сделай пожалуйста если не сложно чтобы файл с конфигом можно было указывать через параметры ну там потом можно не особо критично, Win7 много кто пользуется потому что многим не нравится win8-win11 с кучей мусора на борту да и самим интерфейсом

**Comment by bol-van** [comment]
> делай пожалуйста если не сложно чтобы файл с конфигом можно было указывать через параметры

Сделал.  @<config_filename>

**Comment by Uefi1** [comment]
> Сделал. @<config_filename>

Ага пытаюсь протестировать добавляю в ярлык @config то есть вот так :
zapret-master\binaries\win64\winws.exe" @config.cfg
И при применении @config.cfg автоматом пропадает из ярлыка, наверное лучше используй что то с - или -- =)

**Comment by bol-van** [comment]
На самом деле работает. Он сохраняет символ @. Проблема только при показе. Он обрывает строку на @
В следующем релизе добавлю что-то другое, сейчас пересобирать все из-за этой мелочевки не буду

**Comment by Uefi1** [comment]
> На самом деле работает. Он сохраняет символ @. Проблема только при показе. Он обрывает строку на @ В следующем релизе добавлю что-то другое, сейчас пересобирать все из-за этой мелочевки не буду

Хотелось бы ещё узнать как правильно накатать zapret на андроид думаю через Termux поставить, но автозапуска при включении/выключении же скорее всего не будет так ?

**Comment by bol-van** [comment]
readme
ctrl+f
android

**Comment by Uefi1** [comment]
> readme ctrl+f android

Нигде не нашёл информации или гайда как устанавливать правильно

**Comment by bol-van** [comment]
Там нет пошаговых инструкций, лишь базовые вещи.
Никакой автоматизации не предусмотрено.
В вашем распоряжении nfqws,tpws, iptables и собственные знания как поместить это в автозапуск.
В readme, кстати, сказано куда помещать для magisk и supersu

**Comment by Uefi1** [comment]
> Там нет пошаговых инструкций, лишь базовые вещи. Никакой автоматизации не предусмотрено. В вашем распоряжении nfqws,tpws, iptables и собственные знания как поместить это в автозапуск. В readme, кстати, сказано куда помещать для magisk и supersu

Если бы ты конечно компилил в .apk вообще бы здорово было бы

**Comment by bol-van** [comment]
Было бы что компилить. Надо с нуля пилить приложение под android. Его нет.
Я этим заниматься не буду. Если кто-то будет - велком. В отдельной репе

**Comment by Uefi1** [comment]
> Было бы что компилить. Надо с нуля пилить приложение под android. Его нет. Я этим заниматься не буду. Если кто-то будет - велком. В отдельной репе

Ну а что ты ленишься то

**Comment by bol-van** [comment]
Ну раз ты трудолюб, иди памажи. Засядь так на месяцок, изучи с нуля программирование на android
Напиши, отладь, и порадуй люд

**Comment by Uefi1** [comment]
> Ну раз ты трудолюб, иди памажи. Засядь так на месяцок, изучи с нуля программирование на android Напиши, отладь, и порадуй люд

Ну я программист сам на Delphi правда и ничего сложного перекомпилировать проект под другую платформу не вижу совсем =)

**Comment by koqwp2** [comment]
Здравствуйте не работает winws.exe что делать

**Comment by bol-van** [comment]
Если ожидание было нажать , и оно заработало, то прекратить пользоваться и забыть

**Comment by koqwp2** [comment]
оно типа открывается ти сразу же закрывается антивируса нету только майкрософт его вырубал все равно не работает

**Comment by bol-van** [comment]
Это не графическая программа, а консольная.
Начните с docs/quick_start_windows.txt

**Comment by garett80** [comment]
Давно открыл для себя vless и прочие протоколы, но вот решил попробовать прямой обход DPI и как-то странно работает, честно сказать. У меня Win7 x64, подменил 2 Dll'ки, как в мануале, запустил preset_russia.cmd - вроде как работает, но выборочно. Открывается rutracker и FB, но при этом ни в какую всякие заблокированные в РФ новостные ресурсы. Duckduckgo еле шевелится. Я прошу прощения, конечно, но для семерки, наверное, проще и дешевле гонять всякие v2ray... И как-то очень уж сложно сделано даже для подготовленного пользователя - не находите? Почему нельзя GUI написать с кнопками, по-человечески? Почему нужно собирать по сусекам библиотеки и вручную все это прикручивать? Особенно "порадовало" blockcheck.cmd - лог зашарашил и надо вручную сидеть ковырять там. Ну если бы я разрабатывал, я бы сделал опять-таки форму, в которой заранее бы фильтровал события по !!!! AVAILABLE !!!, например. Кому охота 100500 строк сидеть прокручивать? Я понимаю, что это энтузиазм и все такое, что вы не обязаны, но я вам подсказал с позиции обычного пользователя. Это гораздо сложнее, чем китайские поделки.

**Comment by bol-van** [comment]
Я еще больше обрадую.
Копипаста оттуда может не сработать. Там думать надо и знать как работает DPI.
Так что бросьте это занятие, цените свое время
